src/error.c

```diff
@@ -0,0 +1,80 @@
+// NOTE: element #1 is the error code and element #2 is a short description of the error case.
+// The description isn't used by the compiler itself but is there for tools to utilize
+// in the future.
+#define ERROR_CODES \
+    ECode(InvalidEscape, "Escape sequence is an invalid Unicode codepoint"), \
+    ECode(InvalidNumericEscape, "Escape sequence is an invalid Unicode codepoint"), \
+    ECode(InvalidCodePoint, "An invalid Unicode codepoint"), \
+    ECode(StringContainsNewline, "A string literal contains a newline"), \
+    ECode(UnexpectedEOF, "An unexpected EOF while parsing a string literal"), \
+    ECode(ExpectedDigit, "Expected a digit"), \
+    ECode(FloatOverflow, "Float literal overflow"), \
+    ECode(IntOverflow, "Integer literal overflow"), \
+    ECode(WrongDoubleQuote, "User entered `“` (0x201c) as a quote instead of ASCII"), \
+
+typedef enum ErrorCode {
+#define ECode(e, s) e##Error
+    ERROR_CODES
+#undef ECode
+} ErrorCode;
+
+typedef struct Error {
+    ErrorCode code;
+    Position pos;
+    char *message;
+} Error;
+
+typedef struct ErrorCollector {
+    u32 errorCount;
+} ErrorCollector;
+
+ErrorCollector errorCollector;
+
+b32 HasErrors() {
+    return errorCollector.errorCount > 0;
+}
+
+#define tempErrorBufferLen KB(4)
+char *temporaryErrorBuffer;
```

Comment:
We could use _Thread_local here for thread safe error message printing. I've used it previously for `scanString`.
https://github.com/kai-language/kai-c/blob/master/src/lexer.c#L287

Reply:
Done. This will be the patch until we have workers with their own buffers and error collectors.